### PR TITLE
chore: Enables all tests for TPF

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Enables all tests for TPF

Link to any related issue(s): CLOUDP-288123

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/testutil/acc/advanced_cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"testing"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertToTPFAttrsMap(t *testing.T) {
+	t.Setenv(config.AdvancedClusterV2SchemaEnvVar, "true")
```
Comment (EspenAlbert) [inline internal/testutil/acc/advanced_cluster_test.go]: Why is this necessary?

File: internal/testutil/acc/advanced_cluster.go [hunk 2]
```diff
@@ -101,3 +103,29 @@ func CheckExistsClusterHandlingRetry(projectID, clusterName string) error {
 		return nil
 	})
 }
+
+func ConvertToTPFAttrsMap(attrsMap map[string]string) {
```
Comment (EspenAlbert) [inline internal/testutil/acc/advanced_cluster.go]: Why not return a new map instead of mutating? Seems cleaner to me

File: internal/testutil/acc/advanced_cluster.go [hunk 3]
```diff
@@ -101,3 +103,29 @@ func CheckExistsClusterHandlingRetry(projectID, clusterName string) error {
...
+			newName = strings.ReplaceAll(newName, k, v)
+		}
+		if newName != name {
+			attrsMap[newName] = attrsMap[name]
+			delete(attrsMap, name)
+		}
+	}
+}
+
+var tpfAttrMapping = map[string]string{
```
Comment (EspenAlbert) [inline internal/testutil/acc/advanced_cluster.go]: nice

File: internal/testutil/acc/advanced_cluster.go [hunk 4]
```diff
@@ -101,3 +103,29 @@ func CheckExistsClusterHandlingRetry(projectID, clusterName string) error {
...
+func ConvertToTPFAttrsMap(attrsMap map[string]string) {
+	if !config.AdvancedClusterV2Schema() {
+		return
+	}
+	names := make([]string, 0, len(attrsMap))
+	for name := range attrsMap {
+		names = append(names, name)
+	}
+	for _, name := range names {
+		newName := name
```
Comment (EspenAlbert) [inline internal/testutil/acc/advanced_cluster.go]: Maybe extract these 3 lines as a function: `func renameToTPF(oldName string) string`. Then we can re-use it in the `prefix` for plural data sources and also in `attrSet` checks

File: internal/testutil/acc/advanced_cluster_test.go (hunk 1 above)
Comment (lantoli) [reply internal/testutil/acc/advanced_cluster_test.go]: you're right that it's confusing, I've moved all tests for TPF helper funcs to the same file so they can be run in GH: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2862/commits/984e868ab19b8755d367d153fae9004a24b4e47b

File: internal/testutil/acc/advanced_cluster.go (hunk 2 above)
Comment (lantoli) [reply internal/testutil/acc/advanced_cluster.go]: that's fair, changed here:  https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2862/commits/4e45b6e500a47a048f6499f115079f6fe746fa6a

File: internal/testutil/acc/advanced_cluster.go (hunk 4 above)
Comment (lantoli) [reply internal/testutil/acc/advanced_cluster.go]: thanks, similar idea done here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2862/commits/0c209edde373c984dd4247e52676e97b64347ff5

File: internal/testutil/acc/advanced_cluster.go (hunk 3 above)
Comment (lantoli) [reply internal/testutil/acc/advanced_cluster.go]: went though another refactor, recommend to look at the latest version

**Review by EspenAlbert — APPROVED** [review]
LGTM. Looks great!
